File: baseandroid/src/main/java/com/ocse/baseandroid/view/LoadingView.java
package com.ocse.baseandroid.view;

import android.app.Dialog;
import android.content.Context;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.animation.Animation;
import android.view.animation.AnimationUtils;
import android.view.animation.LinearInterpolator;
import android.widget.ImageView;

import com.ocse.baseandroid.R;


public class LoadingView extends Dialog {
    Context context;

    public LoadingView(Context context) {
        super(context, R.style.dialog);
        this.context = context;
    }


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        View view= LayoutInflater.from(context).inflate(R.layout.loading_view,null);
        setContentView(view);
        ImageView imageView = view.findViewById(R.id.image);
        Animation a = AnimationUtils.loadAnimation(context, R.anim.rotate);
        LinearInterpolator lin = new LinearInterpolator();
        a.setInterpolator(lin);
        imageView.startAnimation(a);
        setCancelable(false);
        setCanceledOnTouchOutside(false);
    }
}

File: baseandroid/src/main/java/com/ocse/baseandroid/view/LoadingView.kt
package com.ocse.baseandroid.view

import android.app.Activity
import android.app.Dialog
import android.content.Context
import android.os.Bundle
import android.view.LayoutInflater
import android.view.animation.AnimationUtils
import android.view.animation.LinearInterpolator
import android.widget.ImageView
import com.ocse.baseandroid.R

class LoadingView(context: Activity) :
    Dialog(context, R.style.dialog) {
    override fun onCreate(savedInstanceState: Bundle) {
        super.onCreate(savedInstanceState)
        val view =
            LayoutInflater.from(context).inflate(R.layout.loading_view, null)
        setContentView(view)
        val imageView =
            view.findViewById<ImageView>(R.id.image)
        val a =
            AnimationUtils.loadAnimation(context, R.anim.rotate)
        val lin = LinearInterpolator()
        a.interpolator = lin
        imageView.startAnimation(a)
        //        setCancelable(false);
        setCanceledOnTouchOutside(false)
    }

    fun setCancelable() {
        setCancelable(true)
        setCanceledOnTouchOutside(true)
    }

}